src/main/scala/net/ruippeixotog/scalascraper/browser/HtmlUnitBrowser.scala

```diff
@@ -123,6 +123,11 @@ object HtmlUnitBrowser {
     def tagName = underlying.getTagName
     def parent = Option(underlying.getParentNode).collect { case elem: DomElement => HtmlUnitElement(elem) }
     def children = underlying.getChildElements.map(HtmlUnitElement)
+    def siblings = {
+      val previousSiblings = Stream.iterate(underlying.getPreviousElementSibling)(_.getPreviousElementSibling).takeWhile(_ != null)
```

Comment:
haha, yeah i actually had it that way originally, but then i was thinking maybe this way would be more efficient? maybe that's just premature optimization though?